src/context/WalletProvider/Ledger/components/Connect.tsx

```diff
@@ -0,0 +1,83 @@
+import { WebUSBLedgerAdapter } from '@shapeshiftoss/hdwallet-ledger-webusb'
+import React, { useState } from 'react'
+import type { RouteComponentProps } from 'react-router-dom'
+import type { ActionTypes } from 'context/WalletProvider/actions'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { KeyManager } from 'context/WalletProvider/KeyManager'
+import { setLocalWalletTypeAndDeviceId } from 'context/WalletProvider/local-wallet'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+import { ConnectModal } from '../../components/ConnectModal'
+import { LedgerConfig } from '../config'
+
+export interface LedgerSetupProps
+  extends RouteComponentProps<
+    {},
+    any // history
+  > {
+  dispatch: React.Dispatch<ActionTypes>
+}
+
+export const LedgerConnect = ({ history }: LedgerSetupProps) => {
+  const { dispatch: walletDispatch, state } = useWallet()
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState<string | null>(null)
+
+  // eslint-disable-next-line no-sequences
+  const setErrorLoading = (e: string | null) => (setError(e), setLoading(false))
+
+  const pairDevice = async () => {
+    setError(null)
+    setLoading(true)
```

Comment:
do we want to handle the case where pairing is already in progress, i.e `isLoading === true` ?

Reply:
Don't believe that can be the case? Since this is triggered by pair button click, and as the button goes in loading state, it will be disabled